python/pyarrow/tests/test_compute_kernels.py

```diff
@@ -0,0 +1,1237 @@
+            try:
+                next_value = next(iterators[idx])
+                parameters_map[constrained_parameter.key] = next_value
+                break
+            except StopIteration:
+                if idx == 0:
+                    return
+                del parameters_map[constrained_parameter.key]
+                del iterators[idx]
+        for reset_idx in range(idx + 1, len(constrained_parameters)):
+            constrained_parameter = constrained_parameters[reset_idx]
+            it = iter(constrained_parameter.sample(parameters_map))
+            iterators.append(it)
+            initial_value = next(it)
+            parameters_map[constrained_parameter.key] = initial_value
+
+
+def get_sample_calls(signature: FunctionSignature):
+    constrained_parameters = signature.parameter_conditions
+    for param_map in sample_combinations_of_types(constrained_parameters):
+        for dynamic_parameter in signature.dynamic_parameters:
+            param_map[dynamic_parameter.key] = dynamic_parameter.compute_type(
+                param_map)
+        args = []
+        has_variadic = False
+        for arg in signature.args:
+            if arg.key == 'Args':
+                # TODO
+                continue
+            data_type = param_map[arg.key]
+            args.append(data_type)
+            has_variadic |= arg.variadic
+        return_type = param_map[signature.output_key]
+        yield args, return_type, param_map.copy()
+        # if has_variadic:
+        #     args.append(args[-1])
+        #     yield args, return_type, param_map.copy()
+
+
+if __name__ == '__main__':
+    for args, return_type in get_sample_calls(function_taxonomy['add'][1]):
+        print(f'{args} -> {return_type}')
+
+
+def get_sample_opts():
+    def assume_timezone_opts(_): return pc.AssumeTimezoneOptions('UTC')
+
+    def extract_regex_opts(_): return pc.ExtractRegexOptions(
+        '(?P<letter>[ab])(?P<digit>\\d)')
+
+    def idx_opts(types): return pc.IndexOptions(pa.scalar(None, type=types[0]))
+
+    def idx_in_opts(types): return pc.SetLookupOptions(
+        value_set=pa.array([None], type=types[0]))
+
+    def match_substr_opts(_): return pc.MatchSubstringOptions('.*')
+
+    def pad_opts(_): return pc.PadOptions(10)
+
+    def partition_opts(_): return pc.PartitionNthOptions(10)
+
+    def replace_slice_opts(_): return pc.ReplaceSliceOptions(0, 2, '  ')
+
+    def replace_substr_opts(_): return pc.ReplaceSubstringOptions('.*', 'foo')
+
+    def select_k_unstable_opts(_): return pc.SelectKOptions(
+        k=3, sort_keys=[("", "ascending")])
+
+    def slice_opts(_): return pc.SliceOptions(0, 10)
+
+    def split_pattern_opts(_): return pc.SplitPatternOptions(pattern='=')
+
+    def strptime_opts(_): return pc.StrptimeOptions(
+        format='%Y-%m-%dT%H:%M:%SZ', unit='ns')
+
+    def trim_opts(_): return pc.TrimOptions(' ')
+
+    return {
+        'ascii_center': pad_opts,
```

Comment:
Many of the functions have an options object that is optional.  Plus, I can't programmatically generate the options objects themselves, just the map.  I think it could be done by putting the objects in a set and then for each function checking if it's options class is in that set but I worry that might be more confusing to piece together later than just having a very clear map.  I'm willing to try it out though.

Reply:
Yes, that's what I meant. It's not a big deal, and I suppose if you forgot to update it for a new kernel you'd notice right away when you ran tests.